lib/nxt_pipeline/step.rb

```diff
@@ -49,7 +56,15 @@ def set_mapped_options(changeset)
     private
 
     attr_writer :result, :status, :error, :mapped_options, :execution_started_at, :execution_finished_at, :execution_duration
-    attr_reader :constructor, :options_mapper
+    attr_reader :constructor, :options_mapper, :pipeline
+
+    def run_callbacks(*args)
+      pipeline.send(:run_callbacks, *args)
```

Comment:
I don't like the fact that we need to call a private method of the pipeline instance using `send` to execute step callbacks.

Reply:
Well it's better than making it public imho.

Reply:
See the comment from my last review of how solve this.